src/averageAges.js

```diff
@@ -52,7 +82,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  function motherFilter(el) {
+    function isMother(element) {
+      return onlyWithSon
+        ? el.name === element.mother && element.sex === 'm'
+        : el.name === element.mother;
+    }
+
+    return people.find(isMother);
+  }
+
+  const mothers = people.filter(motherFilter);
+  let totalChildNumber = 0;
+
+  function callback(acc, curV) {
```

Comment:
bad name

Reply:
good job, but if you split logic from this "callback" you can simplify your code.
hint no need to look mothers just fin all children who have mothers